provider/serialization/casts.go

```diff
@@ -0,0 +1,179 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright 2024 FeatureForm Inc.
+//
+
+package serialization
+
+import (
+	"fmt"
+	"strconv"
+)
+
+func CastNumberToFloat32(value any) (float32, error) {
+	// I have to do one type per case for this to work properly.
+	switch typed := value.(type) {
+	case int:
+		return float32(typed), nil
+	case int32:
+		return float32(typed), nil
+	case int64:
+		return float32(typed), nil
+	case int8:
+		return float32(typed), nil
+	case int16:
+		return float32(typed), nil
+	case float32:
+		return typed, nil
+	case float64:
+		return float32(typed), nil
+	case string:
+		f64, err := strconv.ParseFloat(typed, 32)
+		if err != nil {
+			return 0, fmt.Errorf("Type error: Expected numerical type and got %T", typed)
```

Comment:
should return the `ParseFloat` error too, so the caller knows the exact parsing error.